blog/2023-11-06-idp-data-preso/index.md

```diff
@@ -0,0 +1,87 @@
+3. What establishes the source of truth for an entity
+
+Embracing GitOps practices, the answer to that last question may sound rather trivial: *GIT, git is obviously the source of truth, since, you know ... GitOps!*
+
+However, while git represents the intended source of truth, truth is actually realized where the resource is deployed, revealing its beloved *resource status*. That is why you may hear people sarcastically refer to git as the *source of hope* in GitOps.
+
+Our current collective of practices reveals that there is no silver bullet when deciding entity representations in a developer portal. What establishes the actual source of truth, from which Backstage entity representations to be drawn, primarily depends on company practices and tools DevOps teams have available to them.
+
+If you operate a Hub and Spoke model, where a single control plane cluster is responsible for handling platform requirements where compute clusters handle the actual workload, the hub cluster could very well become the data source for the Backstage entities. On the other hand, if you operate a federated environment where control plane and data plane workloads are scattered across multiple clusters, Backstage could become the unifier that implements custom entity providers to pull and consolidate data from multiple data sources (i.e. the set of clusters with the right data). In a third approach, the CI may take on the job of hydrating entity definitions with metadata and status information it collects from several data sources, eventually pushing the constructed entity to another repository where it can be observed.
+
+Next we discuss these approaches in more details.
+
+## Use your existing CI/CD pipelines to construct the source of truth
+Starting from the specification of entities and infrastructure resources in a
+git repository, this approach utilizes tasks in the CI to hydrate entities with
+information on entity relations, extra metadata, and status of deployed
+resources. To avoid conflating user changes and automated CI changes, the
+hydrated entities are often kept in a separate git repository that mirrors and
+expands entities in the original git repository with intended application
+specifications.
+
+On the positive side:
+
+* This is a relatively simple approach and works for smaller teams with smaller number of applications or systems
+* Having a second git repository to capture the end state of an entity stays closer to the core GitOps practices
+* Does not require significant modification to the developer portal
+
+On the negative side:
+
+* There is inherent duplications that are happening
+* Adding custom metadata by application teams is not as trivial as it requires making changes to the integration workflow, thus bringing load and demand to the DevOps teams
+* Less abstraction in place as end application users are directly exposed to the yaml specification of the entities
```

Comment:
Should users be developers here?

Reply:
I do think that PMs etc; would be end users of the Developer Portal potentially.